tests/std/tests/P0896R4_ranges_alg_generate/test.cpp

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <algorithm>
+#include <cassert>
+#include <concepts>
+#include <ranges>
+
+#include <range_algorithm_support.hpp>
+
+using namespace std;
+
+struct instantiator {
+    template <ranges::output_range<const int&> Out>
+    static constexpr void call() {
+        using ranges::generate, ranges::iterator_t;
+
+        const auto iota_gen = [count = 0]() mutable { return count++; };

```

Comment:
Now that this test contains a `using`-directive, I feel compelled to comment that this init-capture shadows the name of an algorithm.
```suggestion
        const auto iota_gen = [val = 0]() mutable { return val++; };
```